internal/clients/idler.go

```diff
@@ -32,10 +41,28 @@ type clusterView struct {
 	AppDNS string
 }
 
+// Jenkins pod state response structs
+// ErrorCode is an integer that clients to can use to compare errors
+type errorCode uint32
+
+type responseError struct {
+	Code        errorCode `json:"code"`
+	Description string    `json:"description"`
+}
+
+type jenkinsInfo struct {
+	State podState `json:"state"`
+}
+
+type statusResponse struct {
+	Data   *jenkinsInfo    `json:"data,omitempty"`
+	Errors []responseError `json:"errors,omitempty"`
+}
+
 // IdlerService provides methods to talk to the idler client
 type IdlerService interface {
-	IsIdle(tenant string, openShiftAPIURL string) (bool, error)
 	UnIdle(tenant string, openShiftAPIURL string) (int, error)
+	State(tenant string, openShiftAPIURL string) (podState, error)
```

Comment:
Let's keep this consistent with idler APIs. I think there it is `Status`, isn't it?

Reply:
yes, but status is meant to return the entire datastructure related to status which could include many attributes not just the state of a pod.